apis/io-engine/protobuf/v1/pool.proto

```diff
@@ -33,14 +34,41 @@ message PoolTypeValue {
 
 // Create pool arguments.
 message CreatePoolRequest {
-  string name = 1;           // name of the pool
-  google.protobuf.StringValue uuid = 2; // optional uuid for the pool to be created
-  repeated string disks = 3; // disk device paths or URIs to be claimed by the pool
-  PoolType pooltype = 4;     // type of the pool
+  // Name of the pool.
+  string name = 1;
+  // Optional uuid for the pool to be created.
+  google.protobuf.StringValue uuid = 2;
+  // Disk device paths or URIs to be claimed by the pool.
+  repeated string disks = 3;
+  // Type of the pool.
+  PoolType pooltype = 4;
   // Default cluster size is 4MiB (4 * 1024 * 1024), input cluster size must be in "bytes" but
   // must be multiple of 1MiB, else default cluster size i.e 4MiB will be considered. Minimum
   // cluster size can be input as 1048576 (1024 * 1024) bytes (1MiB)
   optional uint32 cluster_size = 5;
+  // Optional pool metadata arguments.
+  optional PoolMetadataArgs md_args = 6;
+}
+
+// Pool metadata arguments.
+message PoolMetadataArgs {
+  // This parameter controls the initial metadata reservation for a storage pool.
+  // This parameter sets a ratio of the desired metadata reservation amount
+  // to the default reservation for the given pool type.
+  //
+  // 1.0 is the default value, and it means to reserve the amount
+  // of disk space for metadata which is default for the pool type.
+  //
+  // Values larger that 1.0 would increase amount of space reserved for metadata.
+  // For example, 2.0 would double the capacity reserved for metadata.
+  //
+  // Values smaller that 1.0 would decrease amount of space reserved for metadata.
+  // For example, 0.5 would half the capacity reserved for metadata.
+  //
+  // The actual amount of disk space in bytes reserved for metadata may
+  // depend on pool size, cluster size, and other pool parameters.
+  // A pool backend may also allow dynamic metadata allocation.
+   optional float md_resv_ratio = 1;
```

Comment:
Does this still need to be optional if the message is already optional?

Reply:
I think yes: to tell the backend to use the default value.